src/main/serverHelper.ts

```diff
@@ -41,23 +41,25 @@ function base64Decode(value: string): string {
  * @param maxPayloadSize
  * @return [value: string, error: string]
  */
-export async function getRequestBody(request: http.IncomingMessage, maxPayloadSize: number): Promise<[string, string]> {
+export async function getRequestBody(request: http.IncomingMessage, maxPayloadSize: number): Promise<[string, string, number]> {
   const chunks: Buffer[] = [];
   let error = '';
+  let errorCode = 200;
   let dataSize = 0;
 
   // Read in the request body
   for await (const chunk of request) {
     dataSize += chunk.length;
-    if (dataSize > maxPayloadSize) {
+    if (dataSize > maxPayloadSize && errorCode === 200) {
```

Comment:
This change seems wrong: you only enter the error branch once, when you are over the `maxPayloadSize`, but in the next loop iteration `errorCode` is no longer `200`, so you fall into the `else` branch and append more data to the `chunks`.

```suggestion
    if (dataSize > maxPayloadSize) {
```

Reply:
Good catch. I'm simplifying the code and adding a comment not to break out of the loop prematurely (because we still need to consume all the input).